knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -748,3 +751,23 @@ func removeCreationTimestamp(manifest *mf.Manifest, _ *serverlessoperatorv1alpha
 	})
 	return err
 }
+
+func injectNamespacedBrokerMonitoring(kk *serverlessoperatorv1alpha1.KnativeKafka) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "ConfigMap" || u.GetName() != "config-namespaced-broker-resources" {
+			return nil
+		}
+		if !openshiftmonitoring.ShouldEnableMonitoring(kk.Spec.Config) {
+			return nil
+		}
```

Comment:
KnativeKafka is not used to configure monitoring for Eventing components, but only KnativeEventing, so we should check on KnativeEventing.

However, I'm fine with a TODO / Jira and address it separately but just to clarify that this won't work as intended.

Reply:
Addressed in [a4225f2](https://github.com/openshift-knative/serverless-operator/pull/1896/commits/a4225f24eee6f7b4545f18b5822046fe47982cdd)